sw/host/opentitanlib/src/transport/ti50emulator/gpio.rs

```diff
@@ -26,47 +26,51 @@ pub enum PinMode {
     OpenDrain,
 }
 
-/// Enumeration representing the GPIO control commands
-/// sent through the socket.
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-#[repr(u8)]
-enum GpioStreamCommand {
-    SetZero = b'0',
-    SetOne = b'1',
-    GetState = b'%',
+/// Structure representing GPIO pin configuration state.
+#[derive(Clone, Copy, Serialize, Deserialize)]
+pub struct GpioConfiguration {
+    pin_mode: PinMode,
+    pull_mode: PullMode,
+    value: bool,
 }
 
-impl From<bool> for GpioStreamCommand {
-    fn from(value: bool) -> GpioStreamCommand {
-        match value {
-            false => GpioStreamCommand::SetZero,
-            true => GpioStreamCommand::SetOne,
+impl GpioConfiguration {
+    pub fn default() -> Self {
+        Self {
+            pin_mode: PinMode::Input,
+            pull_mode: PullMode::None,
+            value: false,
         }
     }
 }
 
-/// Enumeration that represents GPIO status updates caused
-/// by a response to a `GetState` command.
+impl fmt::Display for GpioConfiguration {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:?} {} {}", self.pin_mode, self.pull_mode, self.value)
```

Comment:
Ni: You could write Display for PinMode

Reply:
I have instead used the gpio::PinMode enum, which already implements `Display`.  And furthermore, this method turns out to not be used at all anymore, so I have removed it.

Reply:
Ok